openeo/rest/datacube.py

```diff
@@ -1469,17 +1469,20 @@ def save_result(self, format: str = "GTiff", options: dict = None) -> 'DataCube'
             }
         )
 
-    def download(self, outputfile: Union[str, pathlib.Path, None] = None, format: str = "GTIFF", options: dict = None):
+    def download(self, outputfile: Union[str, pathlib.Path, None] = None, format: str = None, options: dict = None):
         """
         Download image collection, e.g. as GeoTIFF.
         If outputfile is provided, the result is stored on disk locally, otherwise, a bytes object is returned.
         The bytes object can be passed on to a suitable decoder for decoding.
 
         :param outputfile: Optional, an output file if the result needs to be stored on disk.
-        :param format: Optional, defaults to "GTIFF", an output format supported by the backend.
+        :param format: Optional, an output format supported by the backend.
         :param options: Optional, file format options
         :return: None if the result is stored to disk, or a bytes object returned by the backend.
         """
+        if format == None:
```

Comment:
would we need a check here that ensures that the `outputfile` string is there?

Reply:
hmm yeah there is indeed a conflict with "If outputfile is provided..., otherwise, a bytes object is returned."

Reply:
I wasn't aware of this "return as bytes" feature

the behavior could be:
- `download(outputfile=None, format=None)` -> GTiff format, return as bytes
- `download(outputfile=None, format=X)`   format `X`, return as bytes
- `download(outputfile="foo.tiff", format=None)` ->  GTiff format, download to file
- `download(outputfile="result", format=X)`   -> format `X`, download to file

Reply:
Does `imagcollectionclient.py` need the same behaviour? It's not documented for that function, but there is no default now either.